Use of unstable Rust feature `is_multiple_of` breaks builds on Rust < 1.87

## Problem

The crate uses `is_multiple_of()` method on unsigned integers, which is an unstable Rust library feature (`unsigned_is_multiple_of`) that was only stabilized in Rust 1.87.0.

This causes compilation failures when building with Rust versions prior to 1.87.0:

```
error[E0658]: use of unstable library feature `unsigned_is_multiple_of`
   --> src/nonparametric/wilcoxon.rs:491:31
    |
491 |     let estimate = if n_pairs.is_multiple_of(2) {
    |                               ^^^^^^^^^^^^^^
    |
    = note: see issue #128101 <https://github.com/rust-lang/rust/issues/128101> for more information
```

## Affected Files

There are 4 occurrences of `is_multiple_of()`:

1. **src/nonparametric/wilcoxon.rs:491** - `n_pairs.is_multiple_of(2)`
2. **src/nonparametric/wilcoxon.rs:595** - `n_walsh.is_multiple_of(2)`
3. **src/modern/mmd.rs** - `distances.len().is_multiple_of(2)`
4. **src/equivalence/wilcoxon_tost.rs** - `n.is_multiple_of(2)`

## Suggested Fix

Replace all occurrences of `.is_multiple_of(2)` with the equivalent stable expression `% 2 == 0`:

```rust
// Before (requires Rust 1.87+)
if n_pairs.is_multiple_of(2) {

// After (works on all Rust versions)
if n_pairs % 2 == 0 {
```

## Impact

This affects any downstream crates or applications that:
- Use Rust < 1.87.0
- Build for targets where the latest stable Rust isn't available (e.g., some CI environments, WASM builds via emscripten)

## References

- Rust tracking issue: https://github.com/rust-lang/rust/issues/128101
- Stabilization PR: https://github.com/rust-lang/rust/pull/131168 (merged for Rust 1.87.0)

--- a/src/equivalence/wilcoxon_tost.rs
+++ b/src/equivalence/wilcoxon_tost.rs
@@ -405,7 +405,7 @@ fn median_sorted(sorted: &[f64]) -> f64 {
     if n == 0 {
         return f64::NAN;
     }
-    if n.is_multiple_of(2) {
+    if n % 2 == 0 {
         (sorted[n / 2 - 1] + sorted[n / 2]) / 2.0
     } else {
         sorted[n / 2]

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,3 +1,6 @@
+// Allow manual modulo checks for Rust < 1.87 compatibility (is_multiple_of was stabilized in 1.87)
+#![allow(clippy::manual_is_multiple_of)]
+
 pub mod categorical;
 pub mod correlation;
 pub mod distributional;

--- a/src/modern/mmd.rs
+++ b/src/modern/mmd.rs
@@ -303,7 +303,7 @@ fn median_heuristic_1d(data: &[f64]) -> f64 {
     // Find median
     distances.sort_by(|a, b| a.partial_cmp(b).unwrap());
     let mid = distances.len() / 2;
-    if distances.len().is_multiple_of(2) {
+    if distances.len() % 2 == 0 {
         (distances[mid - 1] + distances[mid]) / 2.0
     } else {
         distances[mid]

--- a/src/nonparametric/wilcoxon.rs
+++ b/src/nonparametric/wilcoxon.rs
@@ -488,7 +488,7 @@ fn mann_whitney_estimate_ci(
     diffs.sort_by(|a, b| a.partial_cmp(b).unwrap());
 
     // Hodges-Lehmann estimate: median of pairwise differences
-    let estimate = if n_pairs.is_multiple_of(2) {
+    let estimate = if n_pairs % 2 == 0 {
         (diffs[n_pairs / 2 - 1] + diffs[n_pairs / 2]) / 2.0
     } else {
         diffs[n_pairs / 2]
@@ -592,7 +592,7 @@ fn wilcoxon_estimate_ci(
     walsh.sort_by(|a, b| a.partial_cmp(b).unwrap());
 
     // Hodges-Lehmann estimate: median of Walsh averages
-    let estimate = if n_walsh.is_multiple_of(2) {
+    let estimate = if n_walsh % 2 == 0 {
         (walsh[n_walsh / 2 - 1] + walsh[n_walsh / 2]) / 2.0
     } else {
         walsh[n_walsh / 2]

